### PR TITLE
SpeedLimits stage 1: U interface and texture loading changed

### PR DESCRIPTION
**Pull request by kvakvs**
* Texture and texture-atlas loading changed: now you can define desired texture size
* RoadDefaults textures commented out will be added and loaded in the following update

### CODE REVIEW

**Review by krzychu124 — APPROVED** [review]
Looks good to me 👍

File: TLM/TLM/U/AtlasBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        /// <param name="atlasSizeHint">Square atlas of this size is created.</param>
+        /// <returns>New UI atlas.</returns>
+        public UITextureAtlas CreateAtlas(string atlasName,
+                                          string loadingPath,
+                                          IntVector2 atlasSizeHint) {
+            string fullName = $"TMPE_U_{atlasName}";
+            UITextureAtlas foundAtlas = TextureUtil.FindAtlas(fullName);
+
+            // If is NOT the same as UI default, means the atlas is already loaded (return cached)
+            if (!System.Object.ReferenceEquals(foundAtlas, UIView.GetAView().defaultAtlas)) {
```
Comment (kianzarrin) [inline TLM/TLM/U/AtlasBuilder.cs:38]: I can understand that there is the issue of confusing this with unity object but if you simply use object. that would work too.
Just a side note. no need to fix.

**Comment by kianzarrin** [comment]
Window has opacity but components have transparency ... a bit inconsistent. I understand fixing this might need a lot of code changes and might be out of the scope of this review. so no pressure.
image

**Comment by kianzarrin** [comment]
I have selected the parking tool. but no parking overlay 
image

**Review by kianzarrin — CHANGES_REQUESTED** [review]
see https://github.com/CitiesSkylinesMods/TMPE/pull/1044#issuecomment-769607015

**Comment by krzychu124** [comment]
> I have selected the parking tool. but no parking overlay
> image
> EDIT: once I rotate the screen the parking overlay reappears
> image
> 
> This is sometimes reproducible and sometimes not. So for example:
> 
> * switch to speed tool -> switch to parking tool -> parking sprites are visible
> * switch to Vehicle restriction tool -> exit Vehicle restriction tool -> switch to parking tool -> parking sprites not visible until I rotate the screen.

Pretty old bug, introduced with camera position caching mechanism.

**Comment by kvakvs** [comment]
@kianzarrin For opacity/transparency, the work you mentioned is already done. I store the opacity/transparency in the same float value, in a struct. Its probably the UI inconsistency here?
@krzychu124 The bug is not introduced in this branch? Happy to fix it, but not all problems at once :)

**Review by kianzarrin — APPROVED** [review]
In that case looks good to me

**Comment by krzychu124** [comment]
@kvakvs I think it was introduced when you've added camera transform cache thingy to speed up rendering of overlays long time ago. If you don't move camera and switch tool with overlays they won't show up until you move it a little bit

**Comment by kvakvs** [comment]
I am slowly refactoring overlays into separate modules one per tool.
Can have this as a TODO item in the final Pull request for Speed Limits if its urgent, or as some high prio bug to do after it

**Comment by krzychu124** [comment]
Yeah I'm waiting for speed limits refactor to begin some work on performance improvements. I plan to improve TM:PE to be able to run it with Unity profiler. I'm not sure yet what changes are necessary to make it work and not crash at start

**Pull request merged**